tests/integration/multicluster/routing.go

```diff
@@ -0,0 +1,116 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicluster
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/features"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+// RoutingTest verifies that traffic resources created in a cluster are reflected in the
+// configs a sidecar receives in multicluster scenarios.
+func RoutingTest(t *testing.T, ns namespace.Instance, pilots []pilot.Instance, feature features.Feature) {
+	framework.NewTest(t).
+		Label(label.Multicluster).
+		Features(feature).
+		Run(func(ctx framework.TestContext) {
+			ctx.NewSubTest("routing").
+				Run(func(ctx framework.TestContext) {
+					vsTmpl := `
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: cluster-%d-vs
+spec:
+  hosts:
+  - server-%d
+  http:
+  - route:
+    - destination:
+        host: server-%d
+    headers:
+      request:
+        add:
+          istio-custom-header: user-defined-value`
+
+					clusters := ctx.Environment().Clusters()
+					builder := echoboot.NewBuilderOrFail(ctx, ctx)
+
+					clients := map[resource.ClusterIndex]*echo.Instance{}
+					servers := map[resource.ClusterIndex]*echo.Instance{}
+
+					for _, cluster := range clusters {
+
+						var client echo.Instance
+						clientRef := &client
+						svcName := fmt.Sprintf("client-%d", cluster.Index())
+						builder = builder.With(clientRef, newEchoConfig(svcName, ns, cluster, pilots))
+						clients[cluster.Index()] = clientRef
+
+						var server echo.Instance
+						serverRef := &server
+						svcName = fmt.Sprintf("server-%d", cluster.Index())
+						builder = builder.With(serverRef, newEchoConfig(svcName, ns, cluster, pilots))
+						servers[cluster.Index()] = serverRef
+
+						vs := fmt.Sprintf(vsTmpl, cluster.Index(), cluster.Index(), cluster.Index())
+						cluster.ApplyConfigOrFail(t, ns.Name(), vs)
```

Comment:
You are applying the config to all clusters, how does this test this feature? It could just be reading from the control plane cluster right?

Reply:
it's a different vs for each cluster.  the VirtualService host is of the form `server-0` for the main cluster and `server-1` for the remote.  client-0 in main cluster call server-0 to verify that the header is appended and vice versa for client-1 in remote.  verifying that the remote cluster vs is reflected in the remote sidecars' config.  Granted, there currently is not a test to verify that when client-1 calls server-0 to verify that it does NOT get the header (meaning remote did not receive a envoy config for a main cluster VirtualService)- if that is the intended behavior.

Reply:
> how are same named resources from different clusters handled? Can a workload cluster create a rule in istio-system and impact all namespaces in all clusters? Can a workload cluster create a rule in X namespace and do something like disable TLS for a service in a different cluster in namespace X? Can a workload cluster create a config with the same name/namespace as a ocnfig in another cluster and override it?

THe current PR is aimed to supoort one central IstioD  only listens to istio resources from 1 remote kubernetes API server

Reply:
If you only listen to 1 remote API server then we don't need this pr, you can just pass --kubeconfig to istiod

Reply:
Got it, Thanks! I am Thinking a way to support this for multi remote clusters.. not good answer yet.  Any suggestions from your side? Thanks.

Reply:
@howardjohn reading through docs in https://docs.google.com/document/d/1r58al52otp1uvB7YFHh8ifA_78hYeyd0hNLq4qPDSUM/edit#.  I think this PR will support at least one of the remote cluster as source of config.

Reply:
We need to decide if we are trying to support ONE remote cluster, in which case we don't need any of these PRs, or if we need to support N remote clusters in which case we need a solution to all the questions I asked above. It seems we keep going back and forth